### DISCUSSION TITLE
HMS XXX-2T

### DISCUSSION THREAD

**Discussion by twitnic** [tbnobody/OpenDTU#1126] [General]
Hat schon jemand die OpenDTU mit einem -2T Wechselrichter getestet?

Bspw. https://www.offgridtec.com/hoymiles-hms-800-2t-microinverter-modulwechselrichter.html (ohne extra WLAN, bald sogar mit WLAN https://www.offgridtec.com/hoymiles-hms-800w-2t-mikrowechselrichter-mit-wifi-integriert.html )

**Comment by spueli99** [comment]
Hi, ja klar.
Hab einen HMS 1600, du benötigst halt das richtige Funkmodul. (CMT2300A) 
Ich nutze zur Zeit eine opendtu fusion.

**Comment by zigit780** [comment]
Hi und was ist mit dem HMS-800W-2T der direkt WLAN an Bord hat? 
Kann der direkt ohne FunkModul betrieben werden oder welche Konfiguration wäre nötig?

**Reply by stefan123t** [reply]
Das NetworkProtocol der built-in DTUBI unterscheidet sich signifikant vom Gen2 (MI) / Gen3 (HM) Protokoll über NRF24L01+ bzw. dem eigentlich gleichen Gen3 (HMS/HMT) Protokoll über CMT2300A.
Wir haben dazu bereits einen Feature Request bei AhoyDTU: https://github.com/lumapu/ahoy/issues/1189 und ein Lösungsansatz hier #1430 mit Hilfe der protobuf Dateien aus der Cloud API / SDK App API.

**Comment by mlan159** [comment]
Würde mich auch interessieren, habe tatsächlich schon den HMS-800W-2T geliefert bekommen obwoh ich den ohne Plan bestellt habe. Ich würde aber gerne die Wifi Version behalten wenn das mit openDTU kompatibel ist

**Reply by turrican944** [reply]
Die Wifi Version wird nicht mit OpenDTU funktionieren, da ist quasi eine Original DTU eingebaut und die WR sprechen direkt mit der Cloud. Eine API um die lokal direkt abzufragen soll es nicht geben. Für mich ist das also nix, weil ich die Cloud nicht will. Habe zwar für meinen HMS1600 einen DTU Stick aber nur um ggf. Netzeinstellungen machen zu können.

**Reply by stefan123t** [reply]
Die HMS-800**W**-2T Modelle mit built-in DTUBI WiFi Modul sprechen das sog. NetworkProtocol mit NetworkCommands, die entsprechende Implementierung in AhoyDTU / OpenDTU steht noch aus. 
Wir sollten aber prinzipiell technisch in der Lage sein, das ggf. über einen weiteren AP/STA auf dem lokalen WLAN Interface des ESP32 abzubilden bzw. hierüber sowohl mit der DTUBI als auch mit dem lokalen WLAN (MQTT Server, etc.) kommunizieren zu können.

**Comment by Gelegenheitsbastler** [comment]
Gibt es hier schon Neuigkeiten? 
Mir wurden ebenfalls die neuen HMS-800W-2T geliefert, obwohl ich die HMS-800 bestellt habe. 
Da ich die Daten sehr gerne über die OpenDTU auslesen und dann weiter verarbeiten möchte, wäre es mir wichtig, dass dies möglich ist.

**Comment by DennisOSRM** [comment]
Es ist noch keine Integration in OpenDTU, aber es gibt Fortschritte. Ich habe einen Prototypen, der die aktuellen Leistungsdaten zwei mal pro Minute abfragen kann, wenn der WR im gleichen WLAN ist, und die Daten direkt an einen MQTT Broker schickt.

**Reply by l0tzi** [reply]
ich habe seit dieser Woche auch einen hms-800w-2t im Einsatz. Kann ich beim Testen unterstützen?

**Reply by w0d4** [reply]
Hi, ich habe auch heute einen 800W-2T bekommen.
Wenn ich irgendwie unterstützen kann, bitte gerne melden.
Helfe gerne testen oder was sonst so benötigt wird.

**Reply by cph0x00** [reply]
Auch ich habe einen HMS-800W-2t. Helfe ebenfalls gerne beim Testen :)

**Reply by DennisOSRM** [reply]
Könnt ja mal hier schauen, dort ist der Code: https://github.com/DennisOSRM/hms-mqtt-publisher

**Reply by w0d4** [reply]
Vielen Dank. Läuft perfekt. Mit eine paar Zeilen config in meiner Home Assistant config ist das ganze auch super gut in Home Assistant integriert.

**Reply by zigit780** [reply]
Hi, wenn ich das richtig verstehe ist quasi eine Abfrage der Daten außerhalb der Hoymile eigenen SW möglich. Die Frage die mich noch beschäftigt ... Muss der WR trotzdem Verbindung ins Internet haben bzw. Verbindung zur Cloud oder schick er trotzdem Daten wenn man Ihn nur im lokalen WLAN belässt?
2. Frage wie absehbar wird es eine OpenDTU-Lösung o.ä. geben?
Steh Grad vor der Frage eine Set mit besagten WR zu erwerben oder nicht.

**Reply by DennisOSRM** [reply]
Der WR muss nicht mit dem Internet verbunden sein. Allerdings musst Du eine Verbindung per WLAN herstellen.

**Reply by zigit780** [reply]
OK das ist schon klar. Aber läuft ja dann logischerweise mit meinen Router der mit dem Internet verbunden ist.
Also ich sag's Mal so hat einer von Euch dann auch den WR im Router nach außen geblockt oder die Adressen die der ansteuert. Sonst geht er ja quasi zwangsläufig online.
Und schickt er dann noch Daten?

**Reply by DennisOSRM** [reply]
Ich hatte ihn schon mal geblockt in dem Sinne, dass der Router alle Pakete weg wirft, die vom WR ins Internet gehen. Da ging dann weiterhin das lokale Abfragen. Ich kann das morgen nach Sonnenaufgang gerne nochmal testen.

**Reply by breaver** [reply]
Hallo zusammen, gibt es hier schon Neuigkeiten? Ich hab auch einen HMS 800 mit eingebautem WiFi erhalten, würde aber gerne openDTU nutzen. Das richtige Funkmodul habe ich bereits. Danke

**Reply by julianlelle** [reply]
@DennisOSRM 

Hallo Dennis,

erst einmal vielen Dank für die tolle Arbeit. Eigentlich wollte ich eine Open DTU nutzen, um nur die aktuellen Daten zu sehen (ob der WR läuft). Leider musste ich dann nach der Beschaffung aller Teile feststellen, dass ich die W-Serie habe und somit der Funk nicht mehr funktioniert.

Nun habe ich mir einen Pi3b besorgt, um das ganze in HA zu sehen.
Leider bin ich aber in dem Gebiet (noch) nicht wirklich erfahren. Vielleicht kannst du mir ja ein paar Tipps geben :-)


1. Kann ich den Pi direkt mit dem Wechselrichter verbinden via W-Lan, damit der WR überhaupt nicht im Netzwerk ist? Ich würde dann den Pi per Kabel an den Router verbinden. Oder macht es mehr sinn, den Port für den WR ins Internet zu sperren (falls möglich mit dem Speedport)?

2. Kann ich die IP Adresse ohne S-Miles App im WR zu setzten?

3. Was benötige ich alles auf dem Pi? Also HA ist klar. Aber dann noch dein Programm und den Docker Hub? Oder ist der Docker Hub in deinem Code schon inegriert?


Entschuldigung für die vielen Fragen, aber das ist alles Neuland für mich.

Vielen Dank schonmal und beste Grüße :-)

**Reply by DennisOSRM** [reply]
Hey, ich hoffe die folgenden Infos helfen Dir weiter:

1) Das kannst Du auf die eine oder andere Weise machen. Es sollte beides klappen. Bei mir sind Inverter und RPi im gleichen Netzwerk. Gesperrt hab ich per Default keine Ports.
2) Ich hab das per App gemacht.
3) Vermutlich solltest Du den docker support installieren. Ich hab bei mir noch InfluxDB, iobroker und Grafana installiert um die Daten zu speichern und zu visualisieren.

Hoffe das hilft Dir erstmal weiter.

**Reply by julianlelle** [reply]
Hi Dennis,

vielen Dank für deine schnelle Rückmeldung.

Dann werde ich es auch mal in mein W-Lan einbinden. Sollte ja keine Sicherheitsbedenken geben. Ich möchte nur nicht, dass übers Internet ungewollt Daten ausgetauscht werden.

Mein Ziel ist es dann, auf dem Pi die Daten auch zu speichern und visualisieren mithilfe von Home Assistent. 
Dann lese ich mich mal in deine aufgelisteten Programme noch ein. Aber so wie ich es verstehe, ist dass dann ähnlich wie HA? Ich brauch für den HA dann lediglich noch einen MQTT broker (z.B. Mosquitto Add-on)
Was ist dann genau die Aufgabe des Dockers?

**Reply by Matthies** [reply]
@DennisOSRM Vielen Dank, dein HMS-Wifi-Addon für den HA hat mir den Tag gerettet. Funktioniert tadellos und bietet viel mehr Daten als meine Notlösung in Form einer zwischengeschaltete Zigbee Meßsteckdose. Die kann jetzt weichen, der Wechselrichter wird "richtig" angeschlossen und die Steckdose ist wieder frei für andere Geräte.

**Comment by Ghanrim** [comment]
Also verstehe ich das richtig das ich wegen hms 800w 2t keine Verbindung bekomme ?
![IMG_0753](https://github.com/tbnobody/OpenDTU/assets/170194431/397f4ec0-d3a1-443d-809b-1393dbe0c62b)
![IMG_0751](https://github.com/tbnobody/OpenDTU/assets/170194431/f179db74-6371-4f1c-92f6-3c27cb3f573c)

**Reply by ghost** [reply]
Hi,
Korrekt.
Das dürfte dir mittlerweile laut deinem Beitrag in #1957 mittlerweile klar sein.

Die WR mit WLAN "...W.." werden nicht unterstützt. Ich glaube hier irgendwo gelesen zu haben, das es Leute gibt, die da drann sind.
Ob es openDTU mal bekommt, können nur die Entwickler sagen.

**Comment by opa-rudi** [comment]
Gibt es diesbezüglich Neuigkeiten? Ich würde auch gern einen Wechselrichter mit integriertem WIFI auslesen.

**Comment by ManuStoessel** [comment]
Hallo liebes OpenDTU Team, ich habe diese aktiv scheinenden Projekte gefunden, die die Wifi Modelle von Hoymiles integrieren. Kennt ihr die schon bzw. gibt es Neuigkeiten zu dem Thema Support von HMS-xxxW Modellen?
https://github.com/suaveolent/hoymiles-wifi
https://github.com/DennisOSRM/hms-mqtt-publisher

**Reply by DennisOSRM** [reply]
Mein Publisher fragt die W Modelle per Wifi ab.
